packages/ux-capture/src/Zone.js

```diff
@@ -19,7 +39,36 @@ function Zone(props) {
 	this.measureName = this.props.name;
 
 	// Create a new `ExpectedMark` for each mark
-	this.marks = this.props.marks.map(markName => {
+	const configuredMarkNames = this.props.elements
+		? // new elements array on zone object
+		this.props.elements.map(element => element.marks).flat()
+		: // legacy with direct marks array on zone object
+		this.props.marks;
+
+	// mark names for elements already on the page after interactive transition
+	let elementsAlreadyOnThePage = [];
+
+	if (this.props.startMarkName === INTERACTIVE_TRANSITION_START_MARK_NAME &&
+		this.props.elements &&
+		Array.isArray(this.props.elements)
```

Comment:
on line 42, we only check for the truthiness of `this.props.elements`--does that also need a check that `this.props.elements` is an array?

Reply:
Yes, you're right, we should check the config input every time we should use it to avoid errors.

Reply:
I changed my mind actually - have a tendency for defensive programming without putting enough structure around it.
Rather than doing run-time checking, we should spend some time defining types for input data instead.